internal/provider/vpcs_data_source.go

```diff
@@ -89,50 +114,41 @@ func (d *vpcsDataSource) Read(ctx context.Context, _ datasource.ReadRequest, res
 			resp.Diagnostics.AddError("Unable to Convert Vpc ID", err.Error())
 			return
 		}
-		vpcState := vpcDataSourceModel{
+		vpcState := vpcDSModel{
 			ID:            types.Int64Value(vpcID),
 			Name:          types.StringValue(vpc.Name),
 			ProvisionedID: types.StringValue(vpc.ProvisionedID),
 			ProjectID:     types.StringValue(vpc.ProjectID),
 			CIDR:          types.StringValue(vpc.CIDR),
-			RegionCode:    types.StringValue(vpc.RegionCode),
 			Status:        types.StringValue(vpc.Status),
 			ErrorMessage:  types.StringValue(vpc.ErrorMessage),
-			Created:       types.StringValue(vpc.Created),
 			Updated:       types.StringValue(vpc.Updated),
+			RegionCode:    types.StringValue(vpc.RegionCode),
+			Created:       types.StringValue(vpc.Created),
 		}
 
-		for _, peeringConn := range vpc.PeeringConnections {
-			peeringConnID, err := strconv.ParseInt(peeringConn.ID, 10, 64)
-			if err != nil {
-				resp.Diagnostics.AddError("Unable to Convert Vpc ID", err.Error())
-				return
-			}
-			peeringConnVpcID, err := strconv.ParseInt(peeringConn.VPCID, 10, 64)
-			if err != nil {
-				resp.Diagnostics.AddError("Unable to Convert Vpc ID", err.Error())
-				return
-			}
-			peerConn := &peeringConnectionModel{
-				ID:           types.Int64Value(peeringConnID),
-				VpcID:        types.Int64Value(peeringConnVpcID),
-				Status:       types.StringValue(peeringConn.Status),
-				ErrorMessage: types.StringValue(peeringConn.ErrorMessage),
-			}
-			for _, peerVpc := range peeringConn.PeerVPCs {
-				peerVpcID, err := strconv.ParseInt(peerVpc.ID, 10, 64)
-				if err != nil {
-					resp.Diagnostics.AddError("Unable to Convert Vpc ID", err.Error())
-					return
+		if len(vpc.PeeringConnections) > 0 {
```

Comment:
does the state have an error without this check?

Reply:
you mean should it return an error if there is no peering conn? answer to this is no, as a vpc might exist without a peering connection

Reply:
sorry I meant why does it need to check if a peering connections exists. I was wondering if Terraform gives you one of those unexpected state errors if you do something this without the check

```go
var pcms []peeringConnectionDSModel                        
for _, pc := range vpc.PeeringConnections {
   ...
}
vpcState.PeeringConnections = pcms
```

Reply:
Ah right, it's identical ! changing it for next PR :)